src/seedu/addressbook/data/person/Address.java

```diff
@@ -7,52 +7,215 @@
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
+	
+	private boolean isPrivate;
+	public Block block;
+	public Street street;
+	public Unit unit;
+	public PostalCode postalCode;
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must follow the format: <BLOCK>, <STREET>, #<FLOOR>-<ROOM>, <POSTALCODE>, where BLOCK, FLOOR, ROOM are integers, and POSTALCODE is a 6 digit integer.";
 
     /**
      * Validates given address.
-     *
+     * 
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+    	if (!isValidAddress(address)) {
+    		throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```

Comment:
delegate validating different parts of the address to the relevant class. That is, for e.g., Block should validate if block part of the address string is formatted correctly.